geowebcache/sqlite/src/main/java/org/geowebcache/sqlite/MbtilesBlobStore.java

```diff
@@ -0,0 +1,538 @@
+                    olData = mbtiles.loadTile(tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1]).getData();
+                }
+                // saving the tile
+                mbtiles.saveTile(gtTile);
+                if (useCreateTime) {
+                    // we need to store this tile create time
+                    putTileCreateTime(connection, tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1], System.currentTimeMillis());
+                }
+                Utils.debug(LOGGER, "Tile saved in file '%s'.", file);
+                if (listeners.isEmpty()) {
+                    // no listeners to update we are done
+                    return;
+                }
+                if (olData == null) {
+                    // this was new tile
+                    listeners.sendTileStored(tile);
+                } else {
+                    // this an update
+                    listeners.sendTileUpdated(tile, olData.length);
+                }
+            } catch (Exception exception) {
+                throw Utils.exception(exception, "Error saving tile '%s' in file '%s'.", tile, file);
+            }
+        });
+    }
+
+    @Override
+    public boolean get(TileObject tile) throws StorageException {
+        Utils.updateLogId("GET-" + tile.toString());
+        File file = fileManager.getFile(tile);
+        Utils.debug(LOGGER, "Tile mapped to file '%s'.", file);
+        initDatabaseFileIfNeeded(file, tile.getLayerName(), tile.getBlobFormat());
+        // do work in readonly mode
+        boolean exists = connectionManager.doWork(file, true, connection -> {
+            // instantiating geotools mbtiles reader
+            MBTilesFile mbtiles = GeoToolsMbtilesUtils.getMBTilesFile(connection, file);
+            try {
+                // loading the tile using geotools reader
+                MBTilesTile gtTile = mbtiles.loadTile(tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1]);
+                if (gtTile.getData() != null) {
+                    // the tile exists lets add the data to the tile object
+                    tile.setBlob(Utils.byteArrayToResource(gtTile.getData()));
+                    Utils.debug(LOGGER, "Tile found on file '%s'.", file);
+                    return true;
+                }
+            } catch (Exception exception) {
+                throw Utils.exception(exception, "Error loading tile '%s' from MBTiles file '%s'.", tile, file);
+            }
+            Utils.debug(LOGGER, "Tile not found on file '%s'.", file);
+            return false;
+        });
+        if (exists && useCreateTime) {
+            // the tile exists and we need to set the create time in the tile object
+            Long createdTime = getTileCreateTime(file, tile.getXYZ()[2], tile.getXYZ()[0], tile.getXYZ()[1]);
+            if (createdTime == null) {
+                // no create time associated with this tile lets assume the current as the create time
+                createdTime = System.currentTimeMillis();
```

Comment:
Changed my mind I will go for the last modified time, this will maintain a consistent behavior between windows and linux.